src/Plugin/Block/FooterBlock.php

```diff
@@ -104,10 +103,10 @@ public function build() {
       ],
     ];
 
-    // By some reasons drush config import does not invalidate cache tags.
-    // So for deployment we should take care about this case and
-    // perform cache invalidation
-    // via the "config:oe_corporate_blocks.data.footer" cache tag.
+    // Drush config import does not seem to invalidate cache tags.
+    // When syncing configuration we should take care of this case and
+    // invalidate the cache via the
+    // "config:oe_corporate_blocks.data.footer" cache tag.
     $this->renderer->addCacheableDependency($build, $config);
```

Comment:
We should not inject the renderer just to use what it could have been a static helper method. Please remove the renderer service and perform this manually. Do not need to add cache to the render array manually, but instead create at the top a CacheableMetadata object and fill it up as you go. And before returning, apply that to the render array. **Same for SiteSwitcherBlock**